### PR TITLE
feat(core): add evmtx signer on convert

### PR DESCRIPTION
**Pull request by lykalabrada**
#### What this PR does / why we need it:
Added signed evmtx inside transfer domain for convert screen.

#### Which issue(s) does this PR fixes?:

Fixes DFC-312, DFC-207

#### Additional comments?:

- Due to CORS issue on localhost, `nonce` and `chainId` are currently hardhcoded in this PR.
- Missing e2e tests

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [ ] No console errors on web
- [ ] Tested on Light mode and Dark mode\*
- [ ] Your UI implementation visually matched the rendered design\*
- [ ] Unit tests\*
- [ ] Added e2e tests\*
- [ ] Added translations\*

### CODE REVIEW

File: mobile-app/app/contracts/TransferDomainV1.json [hunk 1]
```diff
@@ -0,0 +1,181 @@
+{
```
Comment (pierregee) [inline mobile-app/app/contracts/TransferDomainV1.json:1]: Could add a section in readme or a comment on where this was taken/generated?

File: docker-compose.yml [hunk 2]
```diff
@@ -12,17 +12,21 @@ services:
     ports:
       - "19553:80"
       - "19552:8080"
-      - "19551:8082"
+      # - "19551:8082"
```
Comment (pierregee) [inline docker-compose.yml:15]: any implication on commenting this?

File: mobile-app/app/api/transaction/transfer_domain.ts [hunk 3]
```diff
@@ -28,44 +33,89 @@ export async function transferDomainSigner(
...
+    chainId: 1133, // (await rpc.getNetwork()).chainId, // TODO: Remove hardcoded data
+    data: data,
+    value: 0,
+    gasLimit: 0,
+    gasPrice: 0,
+    type: 0,
+  };
+
+  const evmtxSigned = (await wallet.signTransaction(tx)).substring(2); // rm prefix `0x`
+  const evmTx = new Uint8Array(Buffer.from(evmtxSigned, "hex") || []);
```
Comment (pierregee) [inline mobile-app/app/api/transaction/transfer_domain.ts:90]: let's extract the evmtx in a new function

**Review by pierregee — CHANGES_REQUESTED** [review]

**Comment by pierregee** [comment]
Also Transfer Domain not working yet on my end

File: mobile-app/app/contracts/TransferDomainV1.json (hunk 1 above)
Comment (lykalabrada) [reply mobile-app/app/contracts/TransferDomainV1.json:1]: Added a [README.md](https://github.com/BirthdayResearch/defichain-wallet/pull/4038/files?short_path=9242be1#diff-9242be105521261c6606959a05d955c59c0a6cd93d166d87256d26e9fa40faf0) inside the `/contracts` folder

File: docker-compose.yml (hunk 2 above)
Comment (lykalabrada) [reply docker-compose.yml:15]: I didn't see any implication yet while performing tests on local playground 👀

**Comment by lykalabrada** [comment]
> Also Transfer Domain not working yet on my end

i was able to replicate this issue, but now it's suddenly working fine on my end. Please try to restart your docker containers if this is still not working for you

File: mobile-app/app/api/transaction/transfer_domain.ts [hunk 4]
```diff
@@ -142,9 +180,81 @@ export function transferDomainCrafter(
...
+  /* TODO: Figure out CORS issue when using the ethRpc */
+  const tx: providers.TransactionRequest = {
+    to: TD_CONTRACT_ADDR,
+    nonce: 3, // await ethRpc.getTransactionCount(from), // TODO: Remove hardcoded data
+    chainId: 1133, // (await rpc.getNetwork()).chainId, // TODO: Remove hardcoded data
+    data: data,
+    value: 0,
+    gasLimit: 0,
+    gasPrice: 0,
+    type: 0,
```
Comment (pierregee) [inline mobile-app/app/api/transaction/transfer_domain.ts:236]: How do we come up with these values?

**Review by pierregee — APPROVED** [review]

**Pull request merged**